testbed/cli/main.py

```diff
@@ -67,7 +70,9 @@ def main(ctx, port, baudrate, edge, devices):
     ctx.ensure_object(dict)
     ctx.obj["port"] = port
     ctx.obj["baudrate"] = baudrate
-    ctx.obj["edge"] = edge
+    ctx.obj["adapter_class"] = (
+        MQTTAdapter if adapter == "mqtt" else SerialAdapter
+    )
```

Comment:
I'm thinking that it would be nice to provide a factory function in the `adapter.py` module. Just drafting something here:

```py
def adapter_class(adapter: str) -> GatewayAdapterBase:
    if adapter == "serial":
         return SerialAdapter
    if adapter == "mqtt":
        return MQTTAdapter
    raise ValueError("Unsupported Adapter type")
```

(I think something even better can be done though)

And here it would become:
```suggestion
    ctx.obj["adapter_class"] = adapter_class(adapter)
```